### DISCUSSION TITLE
Search for existing address in Cuxhaven (in D) shows a postbox in a village 25 km away

### DISCUSSION THREAD

**Discussion by Rainer-o** [osm-search/Nominatim#2673] [Q&A]
Search for "Cuxhaven Dorfstraße 17" returns as result "Briefkasten [Dorfstraße 17, 21782 Bülkau, Dorf, Bülkau, Samtgemeinde Land Hadeln, Landkreis Cuxhaven, Niedersachsen, 21782, Deutschland](https://www.openstreetmap.org/node/4482708665)". This seems a bit strange, as a house with the address "Cuxhaven Dorfstraße 17" exists in OSM.
However the search for "27476 Cuxhaven Dorfstraße 17" returns 2 results, where "Einfamilienhaus [17, Dorfstraße, Stickenbüttel, Cuxhaven, Landkreis Cuxhaven, Niedersachsen, 27476, Deutschland](https://www.openstreetmap.org/way/232742230)" is the first one and the postbox in Bülkau is the second. see also screenshots.
![Screenshot_Cuxhaven_Dorfstraße17_falsch_in_Bülau_Nominatim](https://user-images.githubusercontent.com/46461713/163670125-661251dc-c943-4b9b-956b-2c77ebcaa903.png)
![Screenshot_27476_Cuxhaven_Dorfstraße17_richtig_in_Cuxhaven_Nominatim](https://user-images.githubusercontent.com/46461713/163670134-1180fa01-fe73-4875-8956-e61cfa875a6c.png)

From OSM data I don't see something wrong. The house has correct address attributes https://www.openstreetmap.org/way/232742230 , the boundary of Cuxhaven seems correct and contains the place node with role label https://www.openstreetmap.org/relation/1518688 .
The postbox has ref=Dorfstraße 17, 21782 Bülkau . A user had changed that to addr: tags, because he was trying to improve the search result. I have reverted that, because postboxes should use ref for their location.

Guessing around and looking for data that could mislead Nominatim, I found that the place node of Bülkau https://www.openstreetmap.org/node/240070038 still has the attribute is_in=Cuxhaven,Lüneburg,Niedersachsen,Bundesrepublik Deutschland,Europe , which is deprecated in regions that are well covered with boundary relations like Germany is. 
Not knowing how Nominatim does the search, I prefer to ask the question rather than try-and-error change tags in OSM to improve the Nominatim search results. Is there anything that can be improved in OSM data or is this an issue for the search algorithm?

**Comment by lonvia** [comment]
The questionable ref tags on the postboxes are throwing Nominatim off. See discussion [here](https://github.com/osm-search/Nominatim/issues/2652#issuecomment-1090070482).
